core-test/src/test/scala/org/bitcoins/core/protocol/blockchain/BlockTest.scala

```diff
@@ -43,4 +45,14 @@ class BlockTest extends BitcoinSUnitTest {
     val time = timeBlockParsing(Block.fromHex(lines))
     assert(time <= 15000)
   }
+
+  it must "have serialization symmetry" in {
+    forAllAsync(BlockchainElementsGenerator.block) { block =>
+      Future {
```

Comment:
This seems like a pattern, perhaps we should make a `forAllInParallel` which takes in a `T => Assertion` and wraps the call in a `Future` rather than having users do it. This will hopefully avoid mix-ups in the future as if something becomes future-ified it will have to be adapted with more care. E.g.
```
forAllInParallel(BlockChainElementsGenerator.block) { block =>
  val result = Block(block.hex) == block
  if (!result) logger.warn("block.hex: " + block.hex)
  assert(result)
}
```